### ISSUE TITLE
tokio-fs: Add support for positioned IO

### ISSUE THREAD

**Issue by g2p** [tokio-rs/tokio#699]
I'd like to use positioned io ([pread / pwrite](http://man7.org/linux/man-pages/man2/pread.2.html)) deferred to a thread pool.  It may be interesting to integrate with one of these:

* https://github.com/vasi/positioned-io
* https://github.com/jmesmon/io-at

My preferred API would take an offset and a length, and return a Stream that concatenates to the exact length.

**Comment by tobz** [comment]
A few things:

- this _sounds_ like a reasonable addition, so you should make a PR so we can see how the actual code might look
- returning a `Stream` would not make sense for a `[u8]`-oriented API

**Comment by g2p** [comment]
Because of the likelyhood of short reads, I think it is better to provide a Stream so that partial data can still be used.

**Comment by tobz** [comment]
A `Stream` implies an unbounded read.  We don't do unbounded reads from files themselves.  That's a higher level abstraction.

**Comment by carllerche** [comment]
I'm going to close this for now due to inactivity.

I believe Tokio's position is that we are fine to add positioned IO as a feature, but there is a lot of other fs work that needs to happen first. Also, we are cautious about adding dependencies on third party crates and we definitely are very conservative about including third party crates in the public API.

Once the other tokio-fs work happens, we can revisit this.

**Comment by Sytten** [comment]
@carllerche Did that work append? Is it time to revisit?
